### PR TITLE
Split the context in two to not depend on the sonata user bundle

### PR DESCRIPTION
**Pull request by goetas**

### CODE REVIEW

**Comment by goetas** [comment]
hi @OskarStark , it you are interested in this type of change, i can fix the conflicts and add some tests.

**Comment by OskarStark** [comment]
Please go ahead 👍🏻🙏💡

**Comment by goetas** [comment]
@OskarStark  all the changes have been done. I did not do any logic-change to the code, i've just moved it into a separate class.
The features that were depending on the user bundle had zero test coverage. Now i've added a test for each behat step.

I think that this is a breaking change and it will require a 2.0 release of this package.

File: tests/Context/SonataAdminUserBundleContextTest.php [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+class SonataAdminUserBundleContextTest extends BaseTestCase
+{
+    /**
+     * @var SonataAdminUserBundleContext
+     */
+    private $context;
+
+    /**
+     * @var MockObject|UserManagerInterface
```
Comment (OskarStark) [inline tests/Context/SonataAdminUserBundleContextTest.php]:
```suggestion
     * @var MockObject&UserManagerInterface
```

?

**Review by OskarStark — COMMENTED** [review]
Thanks, one minor.

We can release a 2.0, no problem

**Comment by OskarStark** [comment]
I pushed a `1.x` branch based on `master`.

Can you please add some CHANGELOG?

Plus, are you willing to join the repo as a contributor? I am not using this lib anymore...

**Comment by goetas** [comment]
> I pushed a 1.x branch based on master.

:+1: 


> Can you please add some CHANGELOG?

Done

> Plus, are you willing to join the repo as a contributor? I am not using this lib anymore...

Sure. We are using it in a project (and I had also one or two feature to add :) )

(i will have to figure out how to fix the cs failures (

**Review by OskarStark — APPROVED** [review]

**Pull request merged**